README.md

```diff
@@ -1,117 +1,104 @@
+  <br>
+  Service Worker IPFS Gateway
+  <br>
+</h1>
+
+<p align="center" style="font-size: 1.2rem;">Decentralizing IPFS Gateways by running hash verification on end user's machine.</p>
+
+<p align="center">
+  <a href="https://ipfs.tech"><img src="https://img.shields.io/badge/project-IPFS-blue.svg?style=flat-square" alt="Official Part of IPFS Project"></a>
+  <a href="https://discuss.ipfs.tech"><img alt="Discourse Forum" src="https://img.shields.io/discourse/posts?server=https%3A%2F%2Fdiscuss.ipfs.tech"></a>
+  <a href="https://matrix.to/#/#ipfs-space:ipfs.io"><img alt="Matrix" src="https://img.shields.io/matrix/ipfs-space%3Aipfs.io?server_fqdn=matrix.org"></a>
+  <a href="https://github.com/ipfs-shipyard/service-worker-gateway/actions"><img src="https://img.shields.io/github/actions/workflow/status/ipfs-shipyard/service-worker-gateway/main.yml?branch=main" alt="ci"></a>
+  <a href="https://github.com/ipfs-shipyard/service-worker-gateway/releases"><img alt="GitHub release" src="https://img.shields.io/github/v/release/ipfs-shipyard/service-worker-gateway?filter=!*rc*"></a>
+</p>
+
+<hr />
+
+## About
+
+This project demonstrates
+the use of [Helia](https://helia.io) (IPFS implementation in JS)
+and the [`verified-fetch` library](https://github.com/ipfs/helia-verified-fetch)
+([Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) for IPFS)
+within a [Service Worker](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API)
+to facilitate direct verified retrieval of content-addressed data.
+
+A Service Worker is registered on the initial page load, and then intercepts HTTP requests
+for content stored on IPFS paths such as `/ipfs/*` (immutable) and
+`/ipns/*` (mutable) and returns
+[`Response`](https://developer.mozilla.org/en-US/docs/Web/API/Response) object
+to the browser.
+
+It functions as an IPFS gateway within the browser, offering enhanced security
+([hash verification](https://docs.ipfs.tech/concepts/content-addressing/)
+happens on end user's machine) and reliability (ability to use multiple sources
+of content-addressed blocks) without reliance on a single HTTP server for  IPFS
+tasks.
+
+
+### Goals
+
+The main goals of this project are:
+
+- **Enhancing the robustness of IPFS-based web hosting by eliminating reliance
+  on a single HTTP backend.**
+  - Tasks such as fetching blocks from IPFS content providers (both
+    peer-to-peer and HTTP), verifying that block hashes match the expected CID,
+    and re-assembling blocks into deserialized bytes that can be rendered by
+    the browser, all happens within the end user's machine.
+- **Reducing the operational costs associated with running a HTTP backend.**
+  - By shifting the majority of data retrieval tasks to the user's browser, the
+    backend hosting a website no longer needs to serve as a conduit for all of
```

Comment:
I feel like this point can be refined a bit.

> backend hosting a website no longer needs to serve as a conduit for all of its data

What do you mean by this?

Reply:
The summary is that if you're serving content from some server over HTTP, and you're charged for bandwidth, this will reduce your costs significantly.

However, someone will still be paying for transmitting the bytes (via trustless gateways) and users should still be standing up their own that way. Serving only the necessary bytes via a trustless-gateway, and having clients do validation, requires much less work on the byte server than parsing & serving IPFS requests via a general IPFS HTTP server.